packages/matter.js/src/protocol/interaction/AttributeDataEncoder.ts

```diff
@@ -0,0 +1,249 @@
+    }
+
+    const { path, schema, payload, dataVersion } = attributeData;
+    return TlvAttributeReport.encodeTlv({ attributeData: { path, data: schema.encodeTlv(payload), dataVersion } });
+}
+
+/** Encodes an EventReportPayload into a TlvStream (used for TlvAny type). */
+export function encodeEventPayload(eventPayload: EventReportPayload): TlvStream {
+    const { eventData, eventStatus } = eventPayload;
+    if (eventData === undefined) {
+        return TlvEventReport.encodeTlv({ eventStatus });
+    }
+
+    const {
+        path,
+        schema,
+        payload,
+        eventNumber,
+        deltaEpochTimestamp,
+        epochTimestamp,
+        deltaSystemTimestamp,
+        systemTimestamp,
+        priority,
+    } = eventData;
+    return TlvEventReport.encodeTlv({
+        eventData: {
+            path,
+            data: schema.encodeTlv(payload),
+            priority,
+            systemTimestamp,
+            deltaSystemTimestamp,
+            deltaEpochTimestamp,
+            epochTimestamp,
+            eventNumber,
+        },
+    });
+}
+
+/** Return if an AttributeReportPayload can be chunked or not. */
+export function canAttributePayloadBeChunked(attributePayload: AttributeReportPayload): boolean {
+    const { attributeData } = attributePayload;
+    if (attributeData === undefined) {
+        return false;
+    }
+    const {
+        schema,
+        payload,
+        path: { listIndex },
+    } = attributeData;
+    return schema instanceof ArraySchema && Array.isArray(payload) && payload.length > 0 && listIndex === undefined;
+}
+
+/** Chunk an AttributeReportPayload into multiple AttributeReportPayloads. */
+export function chunkAttributePayload(attributePayload: AttributeReportPayload): AttributeReportPayload[] {
+    const { attributeData } = attributePayload;
+    if (attributeData === undefined) {
+        // Should never happen, but we can make sure it do not break
+        return [attributePayload];
+    }
+    const { schema, path, dataVersion, payload } = attributeData;
+    if (!(schema instanceof ArraySchema) || !Array.isArray(payload)) {
+        // Should never happen, but we can make sure it do not break
+        return [attributePayload];
+    }
+    const chunks = new Array<AttributeReportPayload>();
+    chunks.push({ attributeData: { schema, path: { ...path, listIndex: undefined }, payload: [], dataVersion } });
+    payload.forEach(element => {
+        chunks.push({
+            attributeData: {
+                schema: schema.elementSchema,
+                path: { ...path, listIndex: null },
+                payload: element,
+                dataVersion,
+            },
+        });
+    });
+
+    // If the path of the initial AttributePayload was not compressed, we should compress the chunked data
+    // TODO Enable once Tag Compression is supported https://github.com/project-chip/connectedhomeip/issues/29359
+    /*
+    if (path.enableTagCompression !== true) {
```

Comment:
Given that you put the time in to implement maybe turning it on could be an option?  Then there'd be a reference implementation ;)

Reply:
We can not turn it on because else no one will accept the payload. In fact it will also be a "by interaction model revision" decision later once it is officially supported. So later we need kind of a "add compression when interaction model revision >=X"